nodejs.org/s2i/bin/assemble

```diff
@@ -47,6 +47,12 @@ if [ ! -z "$NPM_MIRROR" ]; then
 	npm config set registry $NPM_MIRROR
 fi
 
+# Restore node modules from the previous build (if they exist).
+if [ "$(ls /tmp/artifacts/ 2>/dev/null)" ]; then
+  echo "---> Restoring build artifacts..."
+  mv /tmp/artifacts/ /opt/app-root/src/test-app/
+fi
+
```

Comment:
Check out the example code [here](https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md#assemble).

```sh
# restore build artifacts
if [ "$(ls /tmp/s2i/artifacts/ 2>/dev/null)" ]; then
    mv /tmp/s2i/artifacts/* $HOME/.
fi
```

s2i places the output of the tar stream from `save-artifacts` in `/tmp/s2i/artifacts`, so that's where you'll want to restore from.